tslearn/metrics.py

```diff
@@ -117,69 +116,8 @@ def _return_path(acc_cost_mat):
     return path[::-1]
 
 
-def njit_cdist_dtw(dataset1, dataset2, global_constraint, sakoe_chiba_radius,
-                   itakura_max_slope):
-    """Compute the cross-similarity matrix between two datasets.
-
-    Parameters
-    ----------
-    dataset1 : array
-        A dataset of time series.
-
-    dataset2 : array
-        Another dataset of time series.
-
-    Returns
-    -------
-    cdist : array
-        Cross-similarity matrix.
-    """
-    n_samples_1 = dataset1.shape[0]
-    n_samples_2 = dataset2.shape[0]
-    cdist_arr = numpy.empty((n_samples_1, n_samples_2))
-    for i in prange(n_samples_1):
-        s1 = dataset1[i]
-        s1 = s1[numpy.isfinite(s1[:, 0])]
-        for j in prange(n_samples_2):
-            s2 = dataset2[j]
-            s2 = s2[numpy.isfinite(s2[:, 0])]
-            mask = compute_mask(s1, s2, global_constraint,
-                                sakoe_chiba_radius, itakura_max_slope)
-            cdist_arr[i, j] = njit_dtw(s1, s2, mask)
-    return cdist_arr
-
-
-def njit_cdist_dtw_self(dataset, global_constraint, sakoe_chiba_radius,
-                        itakura_max_slope):
-    """Compute the cross-similarity matrix between a dataset and itself.
-
-    Parameters
-    ----------
-    dataset : array
-        A dataset of time series.
-
-    Returns
-    -------
-    cdist : array
-        Cross-similarity matrix.
-
-    """
-    n_samples = dataset.shape[0]
-    cdist = numpy.zeros((n_samples, n_samples))
-    for i in prange(n_samples):
-        s1 = dataset[i]
-        s1 = s1[numpy.isfinite(s1[:, 0])]
-        for j in prange(i + 1, n_samples):
-            s2 = dataset[j]
-            s2 = s2[numpy.isfinite(s2[:, 0])]
-            mask = compute_mask(s1, s2, global_constraint,
-                                sakoe_chiba_radius, itakura_max_slope)
-            cdist[i, j] = cdist[j, i] = njit_dtw(s1, s2, mask)
-    return cdist
-
-
-def dtw_path(s1, s2, global_constraint=None,
-             sakoe_chiba_radius=1, itakura_max_slope=2.):
+def dtw_path(s1, s2, global_constraint=None, sakoe_chiba_radius=None,
```

Comment:
Didn't find the line in the diff to comment on, but you can add None as a key to the dictionary as well. They would make the following block redundant in both `dtw_path` and `dtw`:

```
    if global_constraint is not None:
        global_constraint_str = global_constraint
    else:
        global_constraint_str = ""
```

Reply:
Good idea! Done (locally, will be pushed soon).